pkg/auth/defaultmanager.go

```diff
@@ -0,0 +1,38 @@
+package auth
+
+import (
+	"errors"
+	"log"
+	"sync"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+)
+
+var mu sync.Mutex
+
+var defaultKeyManager *KeyManager
+
+// InitializeDefaultKeyManager creates the default manager if it has not created yet.
+// This function must be called in main to make sure the default manager is created during service startup.
+// It will try to create the default manager only once even if called multiple times.
+func InitializeDefaultKeyManager(logger *log.Logger, config *configuration.Registry) (*KeyManager, error) {
```

Comment:
I think it's a good practice to create specific configuration interfaces and pass them instead of using configuration struct pointers.
Then it's clear what you actually need in the function. You don't need the entire configuration. You need only one or a few things from it.
See https://github.com/fabric8-services/fabric8-common/blob/master/auth/token.go#L58 and https://github.com/fabric8-services/fabric8-common/blob/c08218d5adf274cd1950db3636e825dc180921cc/auth/token.go#L70:6 as an example.

It also gives more flexibility in tests.

Reply:
Agreed, it's particularly useful for testing, indeed

Reply:
Ah, yes, GitHub had this hidden. Fixed in c748887.